.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: Run tests
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: test
+    runs-on: ubuntu-18.04
```

Comment:
using `ubuntu-18.04` instead of `ubuntu-latest` so we can run tests over Python 3.4. This is a pure migration of what we already have in travis but we can discuss further if we should still be supporting versions that are deprecated (< 3.7)

Reply:
Great point. I'm fine dropping Python 3.4 support in testing. It's been EOL for quite awhile now.

Let's change to `ubuntu-latest` and use a newer Python version. Maybe `3.7` and `3.9` for testing?

Reply:
that sounds good to me, should I also drop support for 3.4, 3.5 and 3.6 in the setup.py as part of this PR?

Reply:
Yeah, that's probably a good plan.